agent/agentssh/forward.go

```diff
@@ -144,12 +159,11 @@ func (h *forwardedUnixHandler) HandleSSHRequest(ctx ssh.Context, _ *ssh.Server,
 			}
 
 			h.Lock()
-			ln2, ok := h.forwards[addr]
-			if ok && ln2 == ln {
-				delete(h.forwards, addr)
+			if ln2, ok := h.forwards[key]; ok && ln2 == ln {
```

Comment:
It _should_ be, but I guess it'd be possible for a `cancel-streamlocal-forward@openssh.com` to arrive followed by a `streamlocal-forward@openssh.com` for the same socket. And if that ever happened, there could be a race to close the new listener. I thought it would be good to keep it as a safety guard for this scenario.